phangsPipeline/handlerImagingChunked.py

```diff
@@ -0,0 +1,1521 @@
+                    if overwrite:
+                        os.system("rm -rf {}".format(fname_dict[img_type]))
+                    else:
+                        logger.warn("Cube name {} already exists. Skipping.".format(fname_dict[img_type]) +
+                                    " Enable overwrite=True to force writing a new version.")
+                        continue
+
+                logger.info("Making {0} from this list of chunked images:".format(fname_dict[img_type]))
+                logger.info(chunk_fname_dict[img_type])
+
+                ia = casaStuff.image()
+                im = ia.imageconcat(outfile=fname_dict[img_type],
+                                    infiles=chunk_fname_dict[img_type])
+                im.done()
+                ia.close()
+
+            # (optional) clean up the per chunk imaging products
+            if remove_chunks:
+                logger.info("Removing chunked imaging products for image type {}".format(root_name))
+                for img_type in list(chunk_fname_dict) + skip_types:
+                    if chunk_fname_dict[img_type] is None or len(chunk_fname_dict[img_type]) == 0:
+                        continue
+                    for chunk_imagename in chunk_fname_dict[img_type]:
+                        os.system("rm -rf {}".format(chunk_imagename))
+
+
+        @CleanCallFunctionDecorator
+        def task_make_dirty_image(
+                self,
+                chunk_num=None,
+                imaging_method='tclean',
+                backup=True,
+                gather_chunks_into_cube=False,
+                remove_chunks=False,
+        ):
+            """
+            Execute the attached clean call with zero iterations to create
+            a dirty image.
+
+            Set backup to True will make a copy of the dirty image as
+            {imagename}_dirty.image
+            """
+
+            chunks_iter = self.return_valid_chunks(chunk_num=chunk_num)
+
+            logger.info("")
+            logger.info("&%&%&%&%&%&%&%&%&%&%&%&%&%")
+            logger.info("Making dirty images for {} chunks:".format(len(chunks_iter)))
+            logger.info(str(chunks_iter))
+            logger.info("&%&%&%&%&%&%&%&%&%&%&%&%&%")
+            logger.info("")
+
+            if not self._dry_run and casa_enabled:
+
+                self._kh.get_imaging_dir_for_target(self.target, changeto=True)
+
+                for ii, chunk_num in enumerate(chunks_iter):
+
+                    # Make the chunk clean call:
+                    this_clean_call = self.task_initialize_clean_call(chunk_num, stage='dirty')
+
+                    logger.info("")
+                    logger.info("&%&%&%&%&%&%&%&%&%&%&%&%&%")
+                    logger.info("Making dirty image for chunk {}:".format(chunk_num))
+                    logger.info(str(this_clean_call.get_param('imagename')))
+                    logger.info("This is {0} out of {1} to be imaged".format(ii, len(chunks_iter)))
```

Comment:
zero indexing makes this info off by one.  Does it matter? Probably not.

Reply:
No but certainly minimizes confusion. fixed!